cpp/arcticdb/version/version_map.hpp

```diff
@@ -537,7 +539,108 @@ class VersionMapImpl {
         folly::collect(key_futs).get();
     }
 
+    /**
+     * Public for testability only.
+     *
+     * @param stream_id symbol to check
+     * @param load_param the load type
+     * @return whether we have a cached entry suitable for the load type, so do not need to go to storage
+     */
+    bool has_cached_entry(const StreamId &stream_id, const LoadParameter& load_param) const {
+        LoadType requested_load_type = load_param.load_type_;
+        util::check(requested_load_type == LoadType::NOT_LOADED
+                    || requested_load_type == LoadType::LOAD_LATEST
+                    || requested_load_type == LoadType::LOAD_DOWNTO
+                    || requested_load_type == LoadType::LOAD_LATEST_UNDELETED
+                    || requested_load_type == LoadType::LOAD_UNDELETED
+                    || requested_load_type == LoadType::LOAD_FROM_TIME
+                    || requested_load_type == LoadType::LOAD_ALL, "Unexpected load type requested {}", requested_load_type);
+
+        load_param.validate();
+        MapType::const_iterator entry_it;
+        if(!find_entry(entry_it, stream_id)) {
+            return false;
+        }
+
+        const timestamp reload_interval = reload_interval_.value_or(
+                ConfigsMap::instance()->get_int("VersionMap.ReloadInterval", DEFAULT_RELOAD_INTERVAL));
+
+        const auto& entry = entry_it->second;
+        if (const timestamp cache_timing = now() - entry->last_reload_time_; cache_timing > reload_interval) {
+            ARCTICDB_DEBUG(log::version(),
+                           "Latest read time {} too long ago for last acceptable cached timing {} (cache period {}) for symbol {}",
+                           entry->last_reload_time_, cache_timing, reload_interval, stream_id);
+
+            return false;
+        }
+
+        LoadType cached_load_type = entry->load_type_;
+
+        switch(cached_load_type) {
+            case LoadType::NOT_LOADED:
+                return requested_load_type == LoadType::NOT_LOADED;
+            case LoadType::LOAD_LATEST:
+                if (requested_load_type == LoadType::NOT_LOADED || requested_load_type == LoadType::LOAD_LATEST) {
+                    return true;
+                }
+                if (requested_load_type == LoadType::LOAD_DOWNTO) {
+                    return loaded_as_far_as_load_until(*entry, load_param);
+                }
```

Comment:
Could add a similar block if `requested_load_type == LoadType::LOAD_FROM_TIME`, although not required to make correct.